.github/workflows/ci.yml

```diff
@@ -4,22 +4,22 @@ on:
   pull_request: {}
 jobs:
   run:
-    runs-on: macos-11
+    runs-on: ${{ matrix.macos }}
     name: Xcode ${{ matrix.xcode }}
     strategy:
       matrix:
-        xcode: ["12.5.1", "13.0"]
+        xcode: ["12.5.1", "13.4.1", "14.3.1"]
+        include:
+        - xcode: "12.5.1"
+          macos: macos-11
+        - xcode: "13.4.1"
+          macos: macos-12
+        - xcode: "14.3.1"
+          macos: macos-13
+    env:
+      DEVELOPER_DIR: /Applications/Xcode_${{ matrix.xcode }}.app/Contents/Developer
     steps:
     - uses: actions/checkout@master
-    - name: Set Xcode
-      run: |
-        echo "Available Xcode versions:"
-        ls /Applications | grep Xcode
-        echo "Choosing Xcode_${{ matrix.xcode }}.app"
-        sudo xcode-select -s /Applications/Xcode_${{ matrix.xcode }}.app
-        xcodebuild -version
-        swift --version
-        swift package --version
```

Comment:
They do the same thing, but `DEVELOPER_DIR` is slightly easier as it avoids us having to make a sudo call to `xcode-select`. 

Happy to revert this if preferred though, I originally did it when I was debugging an issue as I thought it might have been related (it wasn't).

Reply:
No this is great, much better. I saw it recently too in https://github.com/yonaskolb/XcodeGen/pull/1374/files